{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4e534c7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a7d77a83",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = ['BRCA', 'CRC', 'CCRCC', 'LUAD', 'OV', 'UCEC']\n",
    "mutations = ['truncating', 'missense', 'synonymous']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5001feb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_pro_rna = pd.read_csv('../data/sig_pro_rna.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "99549bb1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>PrologFC</th>\n",
       "      <th>ProAveExpr</th>\n",
       "      <th>ProFDR</th>\n",
       "      <th>RNAlogFC</th>\n",
       "      <th>RNAAveExpr</th>\n",
       "      <th>RNAFDR</th>\n",
       "      <th>diffLogFC</th>\n",
       "      <th>diffAveExpr</th>\n",
       "      <th>cancer</th>\n",
       "      <th>mutation</th>\n",
       "      <th>lrt</th>\n",
       "      <th>ispsQTL</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>332</td>\n",
       "      <td>CCDC59</td>\n",
       "      <td>0.617003</td>\n",
       "      <td>0.386270</td>\n",
       "      <td>0.986818</td>\n",
       "      <td>1.452143</td>\n",
       "      <td>2.956698</td>\n",
       "      <td>0.009917</td>\n",
       "      <td>-0.835139</td>\n",
       "      <td>-2.570427</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>343</td>\n",
       "      <td>CCSER2</td>\n",
       "      <td>0.452063</td>\n",
       "      <td>0.180476</td>\n",
       "      <td>0.986818</td>\n",
       "      <td>2.179300</td>\n",
       "      <td>2.992272</td>\n",
       "      <td>0.002613</td>\n",
       "      <td>-1.727238</td>\n",
       "      <td>-2.811796</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>713</td>\n",
       "      <td>EP300</td>\n",
       "      <td>-0.282211</td>\n",
       "      <td>-0.127549</td>\n",
       "      <td>0.986818</td>\n",
       "      <td>-2.016249</td>\n",
       "      <td>3.394311</td>\n",
       "      <td>0.036776</td>\n",
       "      <td>1.734038</td>\n",
       "      <td>-3.521860</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>879</td>\n",
       "      <td>GLOD4</td>\n",
       "      <td>1.700418</td>\n",
       "      <td>0.055922</td>\n",
       "      <td>0.986818</td>\n",
       "      <td>1.823098</td>\n",
       "      <td>3.517731</td>\n",
       "      <td>0.010391</td>\n",
       "      <td>-0.122680</td>\n",
       "      <td>-3.461809</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1174</td>\n",
       "      <td>LGALS9</td>\n",
       "      <td>7.231326</td>\n",
       "      <td>-0.293448</td>\n",
       "      <td>0.005833</td>\n",
       "      <td>3.627117</td>\n",
       "      <td>3.189487</td>\n",
       "      <td>0.033354</td>\n",
       "      <td>3.604209</td>\n",
       "      <td>-3.482935</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0    Gene  PrologFC  ProAveExpr    ProFDR  RNAlogFC  RNAAveExpr  \\\n",
       "0         332  CCDC59  0.617003    0.386270  0.986818  1.452143    2.956698   \n",
       "1         343  CCSER2  0.452063    0.180476  0.986818  2.179300    2.992272   \n",
       "2         713   EP300 -0.282211   -0.127549  0.986818 -2.016249    3.394311   \n",
       "3         879   GLOD4  1.700418    0.055922  0.986818  1.823098    3.517731   \n",
       "4        1174  LGALS9  7.231326   -0.293448  0.005833  3.627117    3.189487   \n",
       "\n",
       "     RNAFDR  diffLogFC  diffAveExpr cancer  mutation    lrt  ispsQTL  overlap  \n",
       "0  0.009917  -0.835139    -2.570427   BRCA  missense  False    False    False  \n",
       "1  0.002613  -1.727238    -2.811796   BRCA  missense  False    False    False  \n",
       "2  0.036776   1.734038    -3.521860   BRCA  missense  False    False    False  \n",
       "3  0.010391  -0.122680    -3.461809   BRCA  missense  False    False    False  \n",
       "4  0.033354   3.604209    -3.482935   BRCA  missense   True    False    False  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_pro_rna.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e7d0f59",
   "metadata": {},
   "source": [
    "# Find all discordant genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c306f849",
   "metadata": {},
   "outputs": [],
   "source": [
    "overlap = sig_pro_rna[sig_pro_rna['overlap'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8d6d5fdc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>PrologFC</th>\n",
       "      <th>ProAveExpr</th>\n",
       "      <th>ProFDR</th>\n",
       "      <th>RNAlogFC</th>\n",
       "      <th>RNAAveExpr</th>\n",
       "      <th>RNAFDR</th>\n",
       "      <th>diffLogFC</th>\n",
       "      <th>diffAveExpr</th>\n",
       "      <th>cancer</th>\n",
       "      <th>mutation</th>\n",
       "      <th>lrt</th>\n",
       "      <th>ispsQTL</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2085</td>\n",
       "      <td>TAOK1</td>\n",
       "      <td>4.423613</td>\n",
       "      <td>-0.176246</td>\n",
       "      <td>0.025076</td>\n",
       "      <td>1.380925</td>\n",
       "      <td>3.241223</td>\n",
       "      <td>0.420269</td>\n",
       "      <td>3.042688</td>\n",
       "      <td>-3.417469</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2155</td>\n",
       "      <td>TNIK</td>\n",
       "      <td>8.618198</td>\n",
       "      <td>-0.149147</td>\n",
       "      <td>0.014347</td>\n",
       "      <td>2.187661</td>\n",
       "      <td>1.984640</td>\n",
       "      <td>0.601864</td>\n",
       "      <td>6.430537</td>\n",
       "      <td>-2.133787</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2368</td>\n",
       "      <td>ZFHX3</td>\n",
       "      <td>-4.240519</td>\n",
       "      <td>0.212571</td>\n",
       "      <td>0.026223</td>\n",
       "      <td>-0.108695</td>\n",
       "      <td>1.469410</td>\n",
       "      <td>0.998688</td>\n",
       "      <td>-4.131824</td>\n",
       "      <td>-1.256839</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>missense</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2532</td>\n",
       "      <td>BLOC1S2</td>\n",
       "      <td>-5.634590</td>\n",
       "      <td>0.363056</td>\n",
       "      <td>0.003934</td>\n",
       "      <td>-0.019437</td>\n",
       "      <td>3.940234</td>\n",
       "      <td>0.999698</td>\n",
       "      <td>-5.615153</td>\n",
       "      <td>-3.577178</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3368</td>\n",
       "      <td>TP53</td>\n",
       "      <td>15.047397</td>\n",
       "      <td>-4.131038</td>\n",
       "      <td>0.003934</td>\n",
       "      <td>1.444938</td>\n",
       "      <td>3.437722</td>\n",
       "      <td>0.998948</td>\n",
       "      <td>13.602460</td>\n",
       "      <td>-7.568761</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0     Gene   PrologFC  ProAveExpr    ProFDR  RNAlogFC  \\\n",
       "5         2085    TAOK1   4.423613   -0.176246  0.025076  1.380925   \n",
       "6         2155     TNIK   8.618198   -0.149147  0.014347  2.187661   \n",
       "11        2368    ZFHX3  -4.240519    0.212571  0.026223 -0.108695   \n",
       "12        2532  BLOC1S2  -5.634590    0.363056  0.003934 -0.019437   \n",
       "16        3368     TP53  15.047397   -4.131038  0.003934  1.444938   \n",
       "\n",
       "    RNAAveExpr    RNAFDR  diffLogFC  diffAveExpr cancer    mutation   lrt  \\\n",
       "5     3.241223  0.420269   3.042688    -3.417469   BRCA    missense  True   \n",
       "6     1.984640  0.601864   6.430537    -2.133787   BRCA    missense  True   \n",
       "11    1.469410  0.998688  -4.131824    -1.256839   BRCA    missense  True   \n",
       "12    3.940234  0.999698  -5.615153    -3.577178   BRCA  synonymous  True   \n",
       "16    3.437722  0.998948  13.602460    -7.568761   BRCA  synonymous  True   \n",
       "\n",
       "    ispsQTL  overlap  \n",
       "5      True     True  \n",
       "6      True     True  \n",
       "11     True     True  \n",
       "12     True     True  \n",
       "16     True     True  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlap.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "244ac91a",
   "metadata": {},
   "source": [
    "# Find all concordant genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a198b62",
   "metadata": {},
   "outputs": [],
   "source": [
    "concordant = pd.read_csv('../data/concordant.csv')\n",
    "concordant.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52954124",
   "metadata": {},
   "source": [
    "# protein wild vs mut data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e47746eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "exp = pd.DataFrame()\n",
    "\n",
    "for index, row in overlap.iterrows():\n",
    "# for index, row in concordant.iterrows():\n",
    "    gene = row['Gene']\n",
    "    mutation = row['mutation']\n",
    "    cancer = row['cancer']\n",
    "    mut = pd.read_csv('../../../Huang_lab_data/PanCancerProteomicsData_HuangLab/'+\n",
    "                          cancer+'.WXS.SomaticVariant.'+mutation+'.txt.gz', sep='\\t')\n",
    "    mut.columns = mut.columns.to_series().str.split(\".Tumor\",1).apply(lambda x: x[0])\n",
    "    exp_pro = pd.read_csv('../../../Huang_lab_data/PanCancerProteomicsData_HuangLab/'+\n",
    "                      cancer+'.proteome.formatted.normalized.tumor.txt.gz', sep='\\t')\n",
    "#     exp_pro = pd.read_csv('../../../Huang_lab_data/PanCancerProteomicsData_HuangLab/'+\n",
    "#                       cancer+'.transcriptome.FPKM.formatted.tumor.txt.gz', sep='\\t')\n",
    "#     exp_pro = np.log2(exp_pro+1) # rna\n",
    "    exp_pro.columns = exp_pro.columns.to_series().str.split(\".Tumor\",1).apply(lambda x: x[0])\n",
    "    samples = mut.loc[gene][mut.loc[gene] != 0] # mutant\n",
    "#     samples = mut.loc[gene][mut.loc[gene] == 0] # wild\n",
    "\n",
    "    for sampleid in samples.index:\n",
    "        if sampleid in exp_pro.columns.tolist():   \n",
    "            exp = pd.Series(exp_pro.loc[gene][sampleid])\n",
    "            exp.index = [gene]\n",
    "            exp['mutation'] = mutation\n",
    "            exp['cancer'] = cancer\n",
    "            exp['sample_id'] = sampleid\n",
    "            exp = pd.concat([exp, exp])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "22f75b18",
   "metadata": {},
   "outputs": [],
   "source": [
    "gen = exp[(exp.index != 'mutation') & (exp.index != 'cancer') & (exp.index != 'sample_id')]\n",
    "\n",
    "mut = exp[exp.index =='mutation']\n",
    "cancer = exp[exp.index =='cancer']\n",
    "sample_id = exp[exp.index =='sample_id']\n",
    "genid = gen.index\n",
    "\n",
    "gen.reset_index(inplace=True, drop=True)\n",
    "mut.reset_index(inplace=True, drop=True)\n",
    "cancer.reset_index(inplace=True, drop=True)\n",
    "sample_id.reset_index(inplace=True, drop=True)\n",
    "\n",
    "expression = pd.concat([gen, mut, cancer, sample_id], axis=1)\n",
    "expression['gene'] = genid\n",
    "expression.columns=['expression', 'mutation', 'cancer', 'sample_id', 'gene']\n",
    "expression.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4bc8c4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "wild_expression = expression "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e761d36",
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_expression = expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ac41d2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "wild_expression['isMut'] = 'wild'\n",
    "mutant_expression['isMut'] = 'mutant'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21cbdc0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_wildVsMut = pd.concat([wild_expression, mutant_expression])\n",
    "exp_wildVsMut.to_csv('../data/wildVSmut/discordant/pro_exp_wildVsMut.csv')\n",
    "# exp_wildVsMut.to_csv('../data/wildVSmut/discordant/rna_exp_wildVsMut.csv')\n",
    "# exp_wildVsMut.to_csv('../data/wildVSmut/concordant/pro_exp_wildVsMut.csv')\n",
    "# exp_wildVsMut.to_csv('../data/wildVSmut/concordant/rna_exp_wildVsMut.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37e603b6",
   "metadata": {},
   "source": [
    "# Merge violin plot data with HGVSp data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "861830c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,36,37,38,39,40,41,42,43,44,45,46,47,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,90,91,92,93,94,95,96,97,98,99,100,101,102,103,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,127) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,89,91,92,93,94,98,102,103,104) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,31,32,33,35,36,37,38,39,40,41,42,44,45,49,62,68,80,81) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,66,67,68,69,70,71,72,73,74,76,78,79,81,85,88,90,91,93,94,98) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,36,37,38,39,40,41,42,43,44,45,46,47,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,100) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,35,36,37,38,39,40,41,42,44,45) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,81) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,127) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,92,100) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,35,36,37,38,39,40,41,42,44,45,68,90) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "/Users/yuqiliu/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,66,67,68,69,70,71,72,73,74,78,79,80,81,84,86,88,94) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "cordance = ['discordant', 'concordant']\n",
    "product = ['pro', 'rna']\n",
    "\n",
    "for c in cordance:\n",
    "    for p in product:\n",
    "        violin_dat = pd.read_csv('../data/wildVSmut/'+cordance+'/'+product+'_exp_wildVsMut.csv')\n",
    "        for mutation in mutations:\n",
    "            for cancer in cancers:\n",
    "                hgvsp = pd.read_csv('../data/HGVSp/' + cancer + mutation + 'HGVSp.csv')\n",
    "                hgvsp.columns = hgvsp.columns.to_series().str.split(\".Tumor\",1).apply(lambda x: x[0])\n",
    "                hgvsp = hgvsp.rename(columns={'Unnamed: 0':'gene'})\n",
    "                hgvsp.index = hgvsp['gene']\n",
    "                hgvsp = hgvsp.drop(['gene'], axis=1)\n",
    "                for sample_id in hgvsp.columns:\n",
    "                    rows = violin_dat[(violin_dat['mutation'] == mutation) & (violin_dat['cancer'] == cancer) & \n",
    "                    (violin_dat['sample_id'] == sample_id)]\n",
    "                    if rows.shape[0] == 0:\n",
    "                        continue\n",
    "                    genes = rows['gene']\n",
    "                    violin_dat.loc[(violin_dat['mutation'] == mutation) & (violin_dat['cancer'] == cancer) & \n",
    "                    (violin_dat['sample_id'] == sample_id), 'hgvsp'] = list(rows.merge(hgvsp, how='left', on='gene')[sample_id])\n",
    "        violin_dat.to_csv('../data/wildVSmut/'+cordance+'/'+product+'_exp_wildVsMut.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c882fbc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
